src/Microsoft.AspNetCore.Razor.Evolution/TagHelperBinderSyntaxTreePass.cs

```diff
@@ -73,6 +73,8 @@ public RazorSyntaxTree Execute(RazorCodeDocument codeDocument, RazorSyntaxTree s
             // Temporary code while we're still using legacy diagnostics in the SyntaxTree.
             errorList.AddRange(errorSink.Errors.Select(error => RazorDiagnostic.Create(error)));
 
+            errorList.AddRange(descriptors.SelectMany(d => d.GetAllDiagnostics()));
```

Comment:
@NTaylorMullen, I assume this was your feedback. I've added a test for this here https://github.com/aspnet/Razor/pull/1071/files#diff-a56a4864f6790f3cccbd5887c5bc4743R302